SmartDeviceLink/SDLSystemCapabilityManager.m

```diff
@@ -243,6 +262,153 @@ - (void)sdl_hmiStatusNotification:(SDLRPCNotificationNotification *)notification
+    return [self sdl_createDisplayCapabilityListFromDeprecatedDisplayCapabilities:rpc.displayCapabilities buttons:rpc.buttonCapabilities softButtons:rpc.softButtonCapabilities];
+}
+#pragma clang diagnostic pop
+
+- (SDLDisplayCapabilities *)sdl_createDeprecatedDisplayCapabilitiesWithDisplayName:(NSString *)displayName windowCapability:(SDLWindowCapability *)windowCapability {
+    SDLDisplayCapabilities *convertedCapabilities = [[SDLDisplayCapabilities alloc] init];
+#pragma clang diagnostic push
+#pragma clang diagnostic ignored "-Wdeprecated"
+    convertedCapabilities.displayType = SDLDisplayTypeGeneric; //deprecated but it is mandatory...
+#pragma clang diagnostic pop
+    convertedCapabilities.displayName = displayName;
+    convertedCapabilities.textFields = [windowCapability.textFields copy];
+    convertedCapabilities.imageFields = [windowCapability.imageFields copy];
+    convertedCapabilities.templatesAvailable = [windowCapability.templatesAvailable copy];
+    convertedCapabilities.numCustomPresetsAvailable = [windowCapability.numCustomPresetsAvailable copy];
+    convertedCapabilities.mediaClockFormats = @[]; // mandatory field but allows empty array
+    convertedCapabilities.graphicSupported = @([windowCapability.imageTypeSupported containsObject:SDLImageTypeDynamic]);
+    
+    return convertedCapabilities;
+}
+
+- (void)sdl_updateDeprecatedDisplayCapabilities {
+    SDLWindowCapability *defaultMainWindowCapabilities = self.defaultMainWindowCapability;
+    NSArray<SDLDisplayCapability *> *displayCapabilityList = self.displays;
+    
+    if (displayCapabilityList == nil || displayCapabilityList.count == 0) {
+        return;
+    }
+    
+    // Create the deprecated capabilities for backward compatibility if developers try to access them
+    self.displayCapabilities = [self sdl_createDeprecatedDisplayCapabilitiesWithDisplayName:self.displays.firstObject.displayName windowCapability:defaultMainWindowCapabilities];
+    self.buttonCapabilities = defaultMainWindowCapabilities.buttonCapabilities;
+    self.softButtonCapabilities = defaultMainWindowCapabilities.softButtonCapabilities;
+}
+
+- (void)sdl_saveDisplayCapabilityListUpdate:(NSArray<SDLDisplayCapability *> *)newCapabilities {
+    NSArray<SDLDisplayCapability *> *oldCapabilities = self.displays;
+    
+    if (oldCapabilities == nil) {
+        self.displays = newCapabilities;
+        [self sdl_updateDeprecatedDisplayCapabilities];
+        return;
+    }
+    
+    SDLDisplayCapability *oldDefaultDisplayCapabilities = oldCapabilities.firstObject;
+    NSMutableArray<SDLWindowCapability *> *copyWindowCapabilities = [oldDefaultDisplayCapabilities.windowCapabilities mutableCopy];
+    
+    SDLDisplayCapability *newDefaultDisplayCapabilities = newCapabilities.firstObject;
+    NSArray<SDLWindowCapability *> *newWindowCapabilities = newDefaultDisplayCapabilities.windowCapabilities;
+    
+    for (SDLWindowCapability *newWindow in newWindowCapabilities) {
```

Comment:
Please add a comment or two to this for loop to describe what's going on here for future reference